foundation_cms/campaigns/models/campaign_page.py

```diff
@@ -0,0 +1,198 @@
+        FieldPanel("body"),
+        MultiFieldPanel(
+            [
+                FieldPanel("signed_header"),
+                FieldPanel("signed_donation_question"),
+                FieldPanel("signed_body"),
+            ],
+            heading="After Signing Content",
+            classname="collapsible",
+        ),
+        MultiFieldPanel(
+            [
+                FieldPanel("share_header"),
+                FieldPanel("share_body"),
+            ],
+            heading="Sharing Step Content",
+            classname="collapsible",
+        ),
+        MultiFieldPanel(
+            [
+                FieldPanel("thank_you_header"),
+                FieldPanel("thank_you_body"),
+                FieldPanel("thank_you_image"),
+            ],
+            heading="Thank You Content",
+            classname="collapsible",
+        ),
+    ]
+
+    translatable_fields = AbstractBasePage.translatable_fields + [
+        TranslatableField("cta"),
+        SynchronizedField("signed_header"),
+        SynchronizedField("signed_donation_question"),
+        SynchronizedField("signed_body"),
+        SynchronizedField("share_header"),
+        SynchronizedField("share_body"),
+        SynchronizedField("thank_you_header"),
+        SynchronizedField("thank_you_body"),
+        SynchronizedField("thank_you_image"),
+    ]
+
+    subpage_types = [
+        "campaigns.CampaignPage",
+        "core.GeneralPage",
+    ]
+
+    # State managed via URL param ?state=&medium=
+    def serve(self, request):
+        state = request.GET.get("state", "start")
+        medium = request.GET.get("medium", "web")
+        user_email = request.GET.get("email", "")
+
+        if request.method == "POST":
+            action = request.POST.get("action")
+
+            if action == "share":
+                return redirect(f"{self.url}?state=sharing&medium={medium}")
+
+            if action == "skip":
+                return redirect(f"{self.url}?state=end&medium={medium}")
+
+        context = self.get_context(request)
+        context.update(
+            {
+                "state": state,
+                "medium": medium,
+                "user_email": user_email,
+            }
+        )
+
+        return render(request, self.get_template(request), context)
+
+    def get_context(self, request, *args, **kwargs):
+        """Override get_context to add latest articles for More Stories section"""
+        context = super().get_context(request, *args, **kwargs)
+
+        petition_cta = None
+        if self.cta:
+            try:
+                petition_cta = Petition.objects.get(id=self.cta.id)
+            except Petition.DoesNotExist:
+                petition_cta = self.cta
+
+        (default_locale, _) = get_default_locale()
+
+        default_articles = (
+            NothingPersonalArticlePage.objects.live()
+            .public()
+            .filter(locale=default_locale)
+            .order_by("-first_published_at")
+        )
```

Comment:
According to the figma, I think we only need the 2 latest articles. Could we update this with a [:2] please?